File: website-orig/models/lane-assignment.ts
class LaneAssignment {
    lane: number;
    car: Car;
    elapsedTime: number;
    position: number;
    points: number;

    static fromPayload(payload: any): LaneAssignment {
        let assignment = new LaneAssignment();

        assignment.lane = payload.Lane;
        assignment.car = Car.fromPayload(payload.Car);
        assignment.elapsedTime = payload.ElapsedTime;
        assignment.position = payload.Position;
        assignment.points = payload.Points;

        return assignment;
    }

    get elapsedTimeSeconds(): number {
        return this.elapsedTime / 1000;
    }

    set elapsedTimeSeconds(value: number) {
        this.elapsedTime = value * 1000;
    }

    static getLaneAssignmentsFromPayload(payload: any): Array<LaneAssignment> {
        let assignments: Array<LaneAssignment> = new Array<LaneAssignment>();
        
        for (let a of payload) {
            assignments.push(LaneAssignment.fromPayload(a));
        }

        return assignments;
    }
}

File: Rlm.Core/Model/LaneAssignment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;
using Rlm.Core;

namespace Rlm.Core
{
    public interface ILaneAssignment
    {
        int Lane { get; }
        ICar Car { get; }
        long ElapsedTime { get; }
        double ScaleSpeed { get; }
        int Position { get; }
        int Points { get; }
    }

    public class LaneAssignment : ILaneAssignment
    {
        public int Lane { get; set; }
        [JsonConverter(typeof(ConcreteConverter<Car>))]
        public ICar Car { get; set; }
        public long ElapsedTime { get; set; }
        public double ScaleSpeed { get; set; }
        public int Position { get; set; }
        public int Points { get; set; }

        public LaneAssignment()
        {

        }
    }
}
